pkg/ruler/ruler_ring.go

```diff
@@ -75,6 +76,7 @@ func (cfg *RingConfig) RegisterFlags(f *flag.FlagSet) {
 	f.DurationVar(&cfg.FinalSleep, "ruler.ring.final-sleep", 0*time.Second, "The sleep seconds when ruler is shutting down. Need to be close to or larger than KV Store information propagation delay")
 	f.IntVar(&cfg.ReplicationFactor, "ruler.ring.replication-factor", 1, "EXPERIMENTAL: The replication factor to use when loading rule groups for API HA.")
 	f.BoolVar(&cfg.ZoneAwarenessEnabled, "ruler.ring.zone-awareness-enabled", false, "EXPERIMENTAL: True to enable zone-awareness and load rule groups across different availability zones for API HA.")
+	f.StringVar(&cfg.TokensFilePath, "ruler.ring.tokens-file-path", "", "EXPERIMENTAL: File path where tokens are stored. If empty, tokens are not stored at shutdown and restored at startup.")
```

Comment:
I am fine with this. Please include it in https://github.com/cortexproject/cortex/blob/e6a4e495ea2430eba8db8b7291b7944f03ab6471/docs/configuration/v1-guarantees.md too

Reply:
Thank you for the review.

Reply:
I wonder why we need to mark this feature as experimental anymore. Thought the initial feature was added long time ago.

Reply:
@yeya24 are you referring to this new flag for rulers or something else? if the former, this feature does not exist in rulers and therefore should be experimental before stabilization?

Reply:
For other components. I think it might be the same for Ruler.
But I am okay to mark this as experimental for now. We can clean it up later in https://github.com/cortexproject/cortex/issues/5922